e2etests/sanity_test.go

```diff
@@ -20,18 +20,29 @@ func checkMatch(t *testing.T, source string, expectedVuln, matchingVuln v1.Vulne
 	if expectedVuln.Metadata == nil {
 		assert.Nil(t, matchingVuln.Metadata, "Expected no metadata for %s but got some", expectedVuln.Name)
 	} else {
-		for _, keys := range [][]string{
+		keySets := [][]string{
 			{source, "CVSSv2", "ExploitabilityScore"},
 			{source, "CVSSv2", "Score"},
 			{source, "CVSSv2", "ImpactScore"},
 			{source, "CVSSv2", "Vectors"},
-			{source, "CVSSv3", "ExploitabilityScore"},
-			{source, "CVSSv3", "Score"},
-			{source, "CVSSv3", "ImpactScore"},
-			{source, "CVSSv3", "Vectors"},
-		} {
-			assert.NotNil(t, deepGet(expectedVuln.Metadata, keys...), "Value for nil for %+v", keys)
-			assert.Equal(t, deepGet(expectedVuln.Metadata, keys...), deepGet(matchingVuln.Metadata, keys...), "Failed for %+v", keys)
+		}
+
+		// When expected vuln has no CVSSv3 data, do not try to compare it.
+		// This was added when NVD stopped returning CVSSv3 data for some vulns which we had test cases for.
```

Comment:
maybe another ticket here while we're at it :D

Reply:
Sure, what would the ticket be for? Rechecking the vulns to see if NVD add the data back?

Reply:
The sanity test will now fail if the data is added back, with an error message that indicates why.